lib/beaker/hypervisor/abs.rb

```diff
@@ -13,15 +13,39 @@ def initialize(hosts, options)
       @resource_hosts = JSON.parse(resource_hosts)
     end
 
+    def connection_preference(host)
+      vmhostname = host[:vmhostname]
+      if vmhostname && host[:hypervisor] == 'abs'
+        @resource_hosts.each do |resource_host|
+          if resource_host['hostname'] == vmhostname
+            engine = resource_host['engine']
+            case engine
+            when /^(vmpooler|nspooler)$/
+              # putting ip last as its not set by ABS
+              return [:vmhostname, :hostname, :ip]
+            else
+              super
+            end
+          end
+        end
+      end
+      super
```

Comment:
@nwolfe yup you are right. We won't call set_ssh_connection_preference(..) and it won't be called.